wormhole-connect/src/utils/wallet/solana.ts

```diff
@@ -69,52 +75,129 @@ export function fetchOptions() {
   };
 }
 
+// This function signs and sends the transaction while constantly checking for confirmation
+// and resending the transaction if it hasn't been confirmed after the specified interval
+// NOTE: The caller is responsible for simulating the transaction and setting any compute budget
+// or priority fee before calling this function
+// See https://docs.triton.one/chains/solana/sending-txs for more information
 export async function signAndSendTransaction(
   request: SolanaUnsignedTransaction<Network>,
   wallet: Wallet | undefined,
   options?: ConfirmOptions,
 ) {
-  if (!wallet || !wallet.signAndSendTransaction) {
-    throw new Error('wallet.signAndSendTransaction is undefined');
+  if (!wallet) throw new Error('Wallet not found');
+  if (!config.rpcs.solana) throw new Error('Solana RPC not found');
+
+  const commitment = options?.commitment ?? 'finalized';
+
+  const unsignedTx = request.transaction.transaction;
+  const connection = new Connection(config.rpcs.solana);
+  const { blockhash, lastValidBlockHeight } =
+    await connection.getLatestBlockhash(commitment);
+
+  if (isVersionedTransaction(unsignedTx)) {
+    const message = TransactionMessage.decompile(unsignedTx.message);
+    message.recentBlockhash = blockhash;
+    message.instructions.push(
+      ...(await createPriorityFeeInstructions(connection, unsignedTx)),
```

Comment:
Does `createPriorityFeeInstructions` end up being an external call? If so, do you thin we can check the returned value before deconstruct and push?

Reply:
`createPriorityFeeInstructions` is in this same file and it will throw if we can't simulate the transaction. It will always return a non-empty array currently. Even if it doesn't the push would be a no-op